core/src/main/java/org/hisp/dhis/android/core/event/internal/EventQueryBundleInternalFactory.kt

```diff
@@ -61,10 +63,21 @@ internal class EventQueryBundleInternalFactory(
             orgUnitByLimitExtractor,
         ) { it?.eventDateDownload() }
 
+        val eventFilters = (params.eventFilters() ?: emptyList()).plus(
+            eventFilterCollectionRepository.byUid().`in`(params.filterUids()).blockingGet(),
+        )
+
+        val programSettingFilters = eventFilterCollectionRepository.byUid()
+            .`in`(programSettings?.specificSettings()?.get(programUid)?.filters()?.map { it.uid() }).blockingGet()
+
         val builder = EventQueryBundle.builder()
+            .eventFilters(eventFilters.takeIf { it.isNotEmpty() } ?: programSettingFilters)
             .commonParams(commonParams)
 
-        return commonHelper.divideByOrgUnits(commonParams.orgUnitsBeforeDivision, commonParams.hasLimitByOrgUnit) {
+        return commonHelper.divideByOrgUnits(
+            commonParams.orgUnitsBeforeDivision,
+            commonParams.hasLimitByOrgUnit,
+        ) {
             builder.orgUnits(it).build()
```

Comment:
In this logic we should support the case where the ASWA has filters defined but the user is not assigned (sharing settings) to any of them. I have the feeling that the list would be empty here and we can't tell if an empty list means no filters defined or filters defined but not assigned.